src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5681,86 +5681,64 @@ class Parser {
         return this->finalize(node, new FunctionExpressionNode(AtomicString(), std::move(options.params), method.get(), popScopeContext(node), isGenerator));
     }
 
-    FunctionExpressionNode* parseGeneratorMethod()
+    PassRefPtr<FunctionExpressionNode> parseGeneratorMethod()
     {
-        // TODO
-        ALLOC_TOKEN(token);
-        this->nextToken(token);
-        this->throwUnexpectedToken(token);
-        RELEASE_ASSERT_NOT_REACHED();
-        /*
         MetaNode node = this->createNode();
-        const isGenerator = true;
-        const previousAllowYield = this->context.allowYield;
+        const bool previousAllowYield = this->context->allowYield;
+        this->expect(LeftParenthesis);
 
-        this->context.allowYield = true;
-        const params = this->parseFormalParameters();
-        this->context.allowYield = false;
-        const method = this->parsePropertyMethod(params);
-        this->context.allowYield = previousAllowYield;
+        pushScopeContext(AtomicString());
+        this->context->allowYield = true;
+        ParseFormalParametersResult formalParameters = this->parseFormalParameters();
+        this->context->allowYield = false;
+        RefPtr<Node> method = this->isolateCoverGrammar(&Parser::parseFunctionSourceElements);
```

Comment:
`parseFunctionSourceElements` is called instead of `parsePropertyMethod` at this point which is different from that of `esprima` parser. Is there any special reason for not calling the `parsePropertyMethod` itselft? (maybe parsing of parameters issue?)

Reply:
The problem is that the arguments should be parsed in a new scopeContext, but `parsePropertyMethod` is not designed for that use case. That's why I've expanded this function.

Reply:
I see. Would you please leave a comment at this point that `parsePropertyMethod` is replaced with the expanded code? Because parsing process of parameter list may be fixed in the future, so we need to mark this point for any possible revision.

Reply:
Sure!